flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -1240,8 +1247,17 @@ private ApplicationReport startAppMaster(
             Thread.sleep(250);
         }
 
-        // since deployment was successful, remove the hook
-        ShutdownHookUtil.removeShutdownHook(deploymentFailureHook, getClass().getSimpleName(), LOG);
+        if (ensureAppDoRun) {
+            if (ClusterEntrypoint.ExecutionMode.DETACHED
+                    .name()
+                    .equals(
+                            flinkConfiguration.get(
+                                    ClusterEntrypoint.INTERNAL_CLUSTER_EXECUTION_MODE))) {
+                logDetachedClusterInformation(applicationId, LOG);
+            }
```

Comment:
Do we have to keep this logic here? It somehow does not seem to belong here.

Reply:
Got it.